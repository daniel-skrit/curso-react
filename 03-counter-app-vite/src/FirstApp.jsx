
import PropTypes from 'prop-types' 
// const newMessage = "Solid Snake";
// const newNumbers = [1,2,3,4,5,6,7,8,9];
// const MessageObjext = {
//   titulo: "Hola",
//   mensaje: "Mensaje"
// };
// export default function FirstApp() {
//   return (
//     <>
//       <h1>{newMessage}</h1>
//       <h2>Numbers: </h2>
//       <li>{newNumbers}</li>
//       <code>{JSON.stringify(MessageObjext)}</code>
//     </>
//   );
// }


// const getResult = (a,b) => {
//   return a + b;
// }

// export default function FirstApp() {
//   return (
//     <>
//       <h1>{getResult(-48,15)}</h1>
//     </>
//   );
// }

export default function FirstApp({title , subtitle, name }) {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle }</p>
      <p>{ name }</p>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  name: PropTypes.string
}

FirstApp.defaultProps ={
  subtitle: "No hay subtitulo",
  name: "Goku"
}