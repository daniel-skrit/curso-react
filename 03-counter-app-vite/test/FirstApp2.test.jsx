import FirstApp from "../src/FirstApp";
import { render, screen } from "@testing-library/react";

describe("", () => {
  const title = "Hola soy un titulo";
  test("debe de hacer match con el snapshot ", () => {
    const {container} = render( <FirstApp title={title}/>);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar "Hola soy un titulo ', () => {
    const {container} = render( <FirstApp title={title}/>);
    // screen.debug();
    expect ( screen.getByText(title)).toBeTruthy();
  })
  
  test('debe de mostrar el titulo en un h1', () => {
     render( <FirstApp title={title}/>);
    expect( screen.getByRole("heading", { level: 1}).innerHTML).toContain(title);
  })
  
});
