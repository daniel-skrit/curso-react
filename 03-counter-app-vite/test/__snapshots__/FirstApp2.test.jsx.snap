// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` debe de hacer match con el snapshot  1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy un titulo
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Goku
  </p>
</div>
`;
